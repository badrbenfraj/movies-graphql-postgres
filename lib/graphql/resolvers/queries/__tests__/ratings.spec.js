const ratings = require('../ratings')
const { db } = require('../../../../adapters/db')

jest.mock('../../../../adapters/db', () => ({
  db: {
    query: jest.fn().mockReturnValue('dbData'),
  },
}))

describe('queries/ratings', () => {
  beforeEach(() => {
    db.query.mockClear()
  })

  it('should get ratings for a movie', () => {
    ratings(null, { movieId: '1' })

    expect(db.query.mock.calls[0][0]).toMatchSnapshot()
  })

  it('should return database data', async () => {
    const response = await ratings(null, { movieId: '1' })

    expect(response).toEqual('dbData')
  })

  it('should handle errors', async () => {
    db.query.mockImplementation(() => Promise.reject('nope'))

    try {
      await ratings(null, { movieId: '1' })
    } catch (e) {
      expect(e.message).toMatch(/nope/)
    }
  })
})
