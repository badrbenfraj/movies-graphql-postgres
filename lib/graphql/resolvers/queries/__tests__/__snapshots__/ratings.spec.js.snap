// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`queries/ratings should get ratings for a movie 1`] = `
"
    SELECT
      r.rating,
      u.name,
      u.email 
    FROM 
      ratings AS r 
      INNER JOIN users AS u ON u.id = r.user_id
    WHERE r.movie_id = 1
  "
`;
